include/windows/osd.h

```diff
@@ -33,7 +33,31 @@ extern "C" {
 #define BIG_ENDIAN 8765
 #define BYTE_ORDER LITTLE_ENDIAN
 
-int socketpair(int af, int type, int protocol, int socks[2]);
+#define OFI_OSD_INIT()								\
+do{										\
+	WORD wsa_version;							\
+	WSADATA data;								\
+	int err;								\
+										\
+	wsa_version = MAKEWORD(2, 2);						\
+										\
+	err = WSAStartup(wsa_version, &data);					\
+	if(err != 0) {								\
+		/* Tell the user that we could not find a usable */		\
+		/* WinSock DLL.                                  */		\
+		return;								\
+	}									\
+										\
+	InitOnceExecuteOnce(&ofi_init_once, ofi_init_once_cb, &ini_lock, 0);	\
+										\
+}while(0)
+
+#define OFI_OSD_FINI()								\
+do{										\
+	WSACleanup();								\
+} while(0)
```

Comment:
ini_lock is windows specific argument, in case if we got OSX specific initialization we will have to add more aruments. for now this is just macro which hides OS specific details, but clear for general logic of initialization